### ISSUE TITLE
has been blocked by CORS policy

### ISSUE THREAD

**Issue by hannesroth88** [microsoft/AzureSearch_JFK_Files#102]
I followed the Readme.
No errors on Running JfkInitializer. When trying to search on the frontend, I get this kind of error in the Browser Console:

Access to fetch at 'https://xyz-search-service-ztmejq5h7ilrc.search.windows.net/indexes/jfkindex/docs?api-version=2019-05-06&search=oswald&searchMode=all&facet=entities,count:20&$count=true&$top=10&scoringProfile=demoBooster&highlight=text' from origin 'https://xyz-site-ztmejq5h7ilrc.azurewebsites.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

Any idea how to resolve that.

confusing was also that i got this error while building with npm run build:prod:

```
ERROR in [at-loader] ./src/pages/detail-page/detail-page.container.tsx:2:10 
    TS2305: Module '"C:/user/projects/repos/ptdocsearch/AzureSearch_JFK_Files/frontend/node_modules/@types/react-router/index"' has no exported member 'withRouter'.
```

```
PS C:\user\projects\repos\ptdocsearch\AzureSearch_JFK_Files\frontend> npm run build:prod

> jfk-files@1.0.0 build:prod C:\user\projects\repos\ptdocsearch\AzureSearch_JFK_Files\frontend
> npm run clean && env-cmd .env cross-env NODE_ENV=production webpack -p --config=webpack.prod.config.js


> jfk-files@1.0.0 clean C:\user\projects\repos\ptdocsearch\AzureSearch_JFK_Files\frontend
> rimraf dist


[at-loader] Using typescript@2.9.2 from typescript and "tsconfig.json" from C:/user/projects/repos/ptdocsearch/AzureSearch_JFK_Files/frontend/tsconfig.json.


[at-loader] Checking started in a separate process...

[at-loader] Checking finished with 1 errors
Hash: ee8b5c233c1d915a19c0
Version: webpack 3.8.1
Time: 33786ms
                             Asset       Size  Chunks                    Chunk Names
                 assets/img/bg.jpg     178 kB          [emitted]
        assets/fonts/jfk-icons.ttf    2.86 kB          [emitted]
       assets/fonts/jfk-icons.woff    2.94 kB          [emitted]
        assets/fonts/jfk-icons.svg    7.88 kB          [emitted]
 assets/img/jfk-files-scenario.png     472 kB          [emitted]  [big]
  assets/svg/azure-search.logo.svg     1.3 kB          [emitted]
    c88dcd84fc705bf3ee29.vendor.js    2.01 MB       0  [emitted]  [big]  vendor
       a84a40f9c8f3a5e49e35.app.js     161 kB       1  [emitted]         app
 132fd44a57d838c61b75.appStyles.js  440 bytes       2  [emitted]         appStyles
  ced751a49d55cd65f077.manifest.js    1.44 kB       3  [emitted]         manifest
      a84a40f9c8f3a5e49e35.app.css    24.1 kB       1  [emitted]         app
132fd44a57d838c61b75.appStyles.css    5.03 kB       2  [emitted]         appStyles
                        index.html    21.5 kB          [emitted]
        android-chrome-192x192.png    6.76 kB          [emitted]
                 favicon-16x16.png  668 bytes          [emitted]
              apple-touch-icon.png    6.21 kB          [emitted]
                       favicon.ico    15.1 kB          [emitted]
        android-chrome-512x512.png    21.1 kB          [emitted]
                 browserconfig.xml  255 bytes          [emitted]
                 favicon-32x32.png    1.09 kB          [emitted]
                mstile-310x310.png    9.55 kB          [emitted]
                mstile-310x150.png    4.57 kB          [emitted]
                mstile-150x150.png    4.25 kB          [emitted]
                mstile-144x144.png    4.36 kB          [emitted]
                  mstile-70x70.png    2.92 kB          [emitted]
                  site.webmanifest  445 bytes          [emitted]
 [242] ./pages/search-page/index.ts 125 bytes {1} [built]
 [415] ./pages/home-page/home-page.container.tsx 2.84 kB {1} [built]
 [419] ./pages/search-page/search-page.container.tsx 8.76 kB {1} [built]
 [663] ./pages/detail-page/index.ts 125 bytes {1} [built]
 [664] ./pages/detail-page/detail-page.container.tsx 3.33 kB {1} [built]
 [666] ./theme/main.scss 432 bytes {2} [built]
 [703] multi ./app.tsx 28 bytes {1} [built]
 [704] ./app.tsx 417 bytes {1} [built]
 [985] ./app.router.tsx 2.59 kB {1} [built]
[1003] ./pages/home-page/index.ts 115 bytes {1} [built]
[1092] ../node_modules/moment/locale ^\.\/.*$ 3.29 kB {0} [optional] [built]
[1461] ./theme/index.ts 32 bytes {1} [built]
[1462] ./theme/theme.ts 1.13 kB {1} [built]
[1463] multi ./theme/main.scss 28 bytes {2} [built]
[1464] multi babel-polyfill material-ui material-ui-icons material-ui-pickers moment paginator qs react react-dom react-router-dom d3 148 bytes {0} [built]
    + 2610 hidden modules

ERROR in [at-loader] ./src/pages/detail-page/detail-page.container.tsx:2:10 
    TS2305: Module '"C:/user/projects/repos/ptdocsearch/AzureSearch_JFK_Files/frontend/node_modules/@types/react-router/index"' has no exported member 'withRouter'.
Child html-webpack-plugin for "index.html":
     1 asset
       [0] ../node_modules/html-webpack-plugin/lib/loader.js!./index.html 21.8 kB {0} [built]
        + 3 hidden modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!theme/main.scss:
                          Asset     Size  Chunks             Chunk Names
              assets/img/bg.jpg   178 kB          [emitted]
     assets/fonts/jfk-icons.ttf  2.86 kB          [emitted]
    assets/fonts/jfk-icons.woff  2.94 kB          [emitted]
     assets/fonts/jfk-icons.svg  7.88 kB          [emitted]
       [0] ../node_modules/css-loader?{"modules":true,"camelCase":true,"importLoaders":1,"localIdentName":"[local]__[name]___[hash:base64:5]"}!../node_modules/resolve-url-loader!../node_modules/sass-loader/lib/loader.js!./theme/main.scss 1.83 kB {0} [built]
       [3] ./assets/img/bg.jpg 63 bytes {0} [built]
       [4] ./assets/fonts/jfk-icons.eot 4.11 kB {0} [built]
       [5] ./assets/fonts/jfk-icons.ttf 72 bytes {0} [built]
       [6] ./assets/fonts/jfk-icons.woff 73 bytes {0} [built]
       [7] ./assets/fonts/jfk-icons.svg 72 bytes {0} [built]
        + 2 hidden modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/detail-page/detail-page.style.scss:
       [0] ../node_modules/css-loader?{"modules":true,"camelCase":true,"importLoaders":1,"localIdentName":"[local]__[name]___[hash:base64:5]"}!../node_modules/resolve-url-loader!../node_modules/sass-loader/lib/loader.js!./pages/detail-page/detail-page.style.scss 1.32 kB {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/search-page.style.scss:
       [0] ../node_modules/css-loader?{"modules":true,"camelCase":true,"importLoaders":1,"localIdentName":"[local]__[name]___[hash:base64:5]"}!../node_modules/resolve-url-loader!../node_modules/sass-loader/lib/loader.js!./pages/search-page/search-page.style.scss 779 bytes {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/home-page/home-page.style.scss:
       [0] ../node_modules/css-loader?{"modules":true,"camelCase":true,"importLoaders":1,"localIdentName":"[local]__[name]___[hash:base64:5]"}!../node_modules/resolve-url-loader!../node_modules/sass-loader/lib/loader.js!./pages/home-page/home-page.style.scss 1.84 kB {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/pagination/pagination.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/spacer/spacer.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/horizontal-separator/horizontal-separator.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/footer/footer.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/graph/graph-view.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/item/item-collection-view.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/page-bar/page-bar.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/home-page/components/caption/caption.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/home-page/components/search/search-button.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/home-page/components/search/search-input.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/search/search.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/drawer/drawer.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/facets/facet-view.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/detail-page/components/toolbar/toolbar.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/logo-jfk/style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/search/autocomplete.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/drawer/drawer-bar.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/page-bar/view-mode-toggler.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/facets/facet-item.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/item/item.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/placeholder/azure-button.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/placeholder/dialog.styles.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/hocr/hocr-document/hocr-document.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/hocr/hocr-preview/hocr-preview.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/hocr/hocr-proofreader/hocr-proofreader.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/placeholder/link.styles.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/facets/facet-body.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/facets/facet-header.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/vertical-separator/vertical-separator.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/menu-button/menu-button.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/hocr/hocr-preview/hocr-page.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/hocr/hocr-preview/hocr-node.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/logo-microsoft/style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/chevron/chevron.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/selection-controls/year-picker.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!pages/search-page/components/selection-controls/checkbox-list.style.scss:
       2 modules
Child extract-text-webpack-plugin ../node_modules/extract-text-webpack-plugin/dist ../node_modules/css-loader/index.js??ref--8-2!../node_modules/resolve-url-loader/index.js!../node_modules/sass-loader/lib/loader.js!common/components/hocr/tooltip/tooltip.style.scss:
       2 modules
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! jfk-files@1.0.0 build:prod: `npm run clean && env-cmd .env cross-env NODE_ENV=production webpack -p --config=webpack.prod.config.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the jfk-files@1.0.0 build:prod script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I could only make the build work with changing from:
`import { withRouter, RouteComponentProps } from "react-router";
`
to

`import { withRouter, RouteComponentProps } from "react-router-dom";`

could the cors error come from my change?

**Comment by Careyjmac** [comment]
Closing due to inactivity

Though from what I can tell, this is a weird transient condition that for some reason CORs don't get set up correctly like they are supposed to. If this repeated after multiple deployment attempts (or really just runs of the initializer) then feel free to open a new issue. As for the front end build issue, this was probably fixed by #105 as I cannot reproduce it now.
